src/components/body/components/chatRoom/components/charArea/chatArea.tsx

```diff
@@ -0,0 +1,75 @@
+//libs
+import { useEffect, useRef } from "react";
+
+//components
+import { TeamMateInfo } from "./components/teamMateInfo/TeamMateInfo";
+// import { Message } from "./components/message/Message";
+import { MessageGroup } from "./components/messageGroup/MessageGroup";
+
+//type
+import { Message as MessageType, User } from "../../../../../../types/types";
+
+//style
+import styles from "./chatArea.module.css";
+
+type ChatAreaProps = {
+  activeTeamMate: User;
+  chat: MessageType[] | undefined;
+};
+
+const groupSameDayMessages = (
+  chat: MessageType[] | undefined,
+  dayArr: boolean[]
+) => {
+  if (!chat) return [];
+
+  const sameDayMessages: MessageType[][] = [];
+
+  let tempArr: MessageType[] = [];
+  dayArr.forEach((ele: boolean, idx: number) => {
+    if (ele && idx) {
+      if (tempArr) sameDayMessages.push(tempArr);
+      tempArr = [];
+    }
+    tempArr.push(chat?.[idx]);
+  });
+  if(tempArr) sameDayMessages.push(tempArr);
+  return sameDayMessages;
+};
+
+export const ChatArea = ({ activeTeamMate, chat }: ChatAreaProps) => {
+  const lastMessageRef = useRef<HTMLDivElement>(null);
+  const dayMap = new Map();
+
+  const dayArr: boolean[] = [];
+  chat?.map((message: MessageType, idx: number) => {
+    dayArr[idx] = !dayMap.has(message?.day);
+    dayMap.set(message?.day, true);
+    return null;
+  });
+
+  const sameDayMessages = groupSameDayMessages(chat, dayArr);
+
+  useEffect(() => {
```

Comment:
Should find a better way to do this.

Every time any of its ancestor components re-renders, this hook will execute.